src/Datagrid/Pager.php

```diff
@@ -441,9 +441,21 @@ public function getLastIndice()
     }
 
     /**
+     * @deprecated since sonata-project/admin-bundle 3.86, use countResults instead
+     *
      * @return int
      */
     public function getNbResults()
+    {
+        @trigger_error(sprintf(
+            'The method "%s()" is deprecated since sonata-project/admin-bundle 3.86 and will be removed in 4.0. Use countResults() instead.',
+            __METHOD__
+        ), E_USER_DEPRECATED);
+
+        return $this->countResults();
+    }
+
+    public function countResults(): int
     {
         return $this->nbResults;
```

Comment:
The property should be renamed too.

Reply:
We cannot rename this. Its `protected` :cry:

Reply:
we can rename + make it private on master I guess

Reply:
See https://github.com/sonata-project/SonataAdminBundle/pull/6732#pullrequestreview-560708661

I think we should just deprecate the property too, and not providing a countResult implementation.

Reply:
deprecated the property as well. So on master we can add a new private property

Reply:
> deprecated the property as well. So on master we can add a new private property

What I recommended was finally to not add the `countResults` in the AbstractPager and let persistence bundle adding both a private property and a method.

For instance the SimplePager is not using the `nbResults` property. And some others Pager could too.

Reply:
Yes now I got what you meant :+1: Makes sense.